packages/components/src/components/FunctionChart.tsx

```diff
@@ -34,41 +41,38 @@ export const FunctionChart: React.FC<FunctionChartProps> = ({
       return result1;
     }
   };
-  let validResult = getValidResult();
-  let resultType = validResult.tag === "Ok" ? validResult.value.tag : "Error";
+  const validResult = getValidResult();
+  const resultType =
+    validResult.tag === "Ok" ? validResult.value.tag : ("Error" as const);
 
-  let component = () => {
-    switch (resultType) {
-      case "distribution":
-        return (
-          <FunctionChart1Dist
-            fn={fn}
-            chartSettings={chartSettings}
-            environment={environment}
-            height={height}
-          />
-        );
-      case "number":
-        return (
-          <FunctionChart1Number
-            fn={fn}
-            chartSettings={chartSettings}
-            environment={environment}
-            height={height}
-          />
-        );
-      case "Error":
-        return (
-          <ErrorBox heading="Error">The function failed to be run</ErrorBox>
-        );
-      default:
-        return (
-          <ErrorBox heading="No Viewer">
-            There is no function visualization for this type of function
-          </ErrorBox>
-        );
-    }
-  };
-
-  return component();
+  switch (resultType) {
+    case "distribution":
+      return (
+        <FunctionChart1Dist
+          fn={fn}
+          chartSettings={chartSettings}
+          environment={environment}
+          height={height}
+        />
+      );
+    case "number":
+      return (
+        <FunctionChart1Number
+          fn={fn}
+          chartSettings={chartSettings}
+          environment={environment}
+          height={height}
+        />
+      );
+    case "Error":
+      return (
+        <ErrorAlert heading="Error">The function failed to be run</ErrorAlert>
+      );
+    default:
+      return (
+        <MessageAlert heading="Function Display Not Supported">
+          There is no function visualization for this type of output
```

Comment:
It would be great to add `resultType` to this error message, just so people know what they returned.